src/queue.ts

```diff
@@ -0,0 +1,1052 @@
+
+    function serialize(message: Message): string {
+      if (codec) {
+        return codec.write(message);
+      } else if (typeof message === "string") {
+        return message;
+      } else if (Buffer.isBuffer(message)) {
+        return message.toString("utf8");
+      } else {
+        throw new Error(
+          `Message must be a string if there is no 'serializer' configured, ${message}`
+        );
+      }
+    }
+
+    this.sendMessage = makeIntegration<
+      "AWS.SQS.SendMessage",
+      (input: SendMessageRequest<Message>) => Promise<AWS.SQS.SendMessageResult>
+    >({
+      kind: "AWS.SQS.SendMessage",
+      native: {
+        bind: (func) => this.resource.grantSendMessages(func.resource),
+        preWarm: (context) => context.getOrInit(SQSClient),
+        call: async ([input], context) => {
+          const sqs = context.getOrInit(SQSClient);
+
+          const messageBody = serialize(input.MessageBody);
+
+          // @ts-ignore
+          delete input.MessageBody;
+
+          const updatedQueueUrl =
+            lambdaQueueUrlRetriever?.(queueUrl) ?? queueUrl;
+
+          const response = await sqs
+            .sendMessage({
+              ...input,
+              MessageBody: messageBody,
+              QueueUrl: updatedQueueUrl,
+            })
+            .promise();
+
+          return response;
+        },
+      },
+      asl: (call, context) => {
+        const input = call.args[0]?.expr;
+
+        if (input === undefined) {
+          throw new SynthError(
+            ErrorCodes.Invalid_Input,
+            `the first argument 'input' is required by sendMessage`
+          );
+        }
+
+        this.resource.grantSendMessages(context.role);
+
+        return context.evalExprToJsonPathOrLiteral(
+          input,
+          (input, { addState }) => {
+            if (ASLGraph.isJsonPath(input)) {
+              addState({
+                Type: "Pass",
+                ResultPath: `${input.jsonPath}.QueueUrl`,
+                Result: this.resource.queueUrl,
+                Next: ASLGraph.DeferNext,
+              });
+              if (this.serializer.dataType === DataType.Json) {
+                // request.MessageBody.value = JSON.stringify(request.MessageBody)
+                addState({
+                  Type: "Pass",
+                  ResultPath: `${input.jsonPath}.MessageBody`,
+                  Parameters: {
+                    "value.$":
+                      "States.JsonToString(${input.jsonPath}.MessageBody)",
+                  },
+                  Next: ASLGraph.DeferNext,
+                });
+                // it's unfortunate that we have to do this
+                // request.MessageBody = request.MessageBody.value
+                addState({
+                  Type: "Pass",
+                  InputPath: `${input.jsonPath}.MessageBody.value`,
+                  OutputPath: `${input.jsonPath}.MessageBody`,
```

Comment:
ResultPath

OutputPath will erase all of the state, almost always use ResultPath